lang/csharp/versions.props

```diff
@@ -0,0 +1,33 @@
+<!--
+   Licensed to the Apache Software Foundation (ASF) under one or more
+   contributor license agreements.  See the NOTICE file distributed with
+   this work for additional information regarding copyright ownership.
+   The ASF licenses this file to You under the Apache License, Version 2.0
+   (the "License"); you may not use this file except in compliance with
+   the License.  You may obtain a copy of the License at
+
+       https://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+-->
+<Project>
+  <PropertyGroup Label="Package Versions">
+    <MicrosoftBuildFrameworkVersion>16.8.0</MicrosoftBuildFrameworkVersion>
+    <MicrosoftBuildUtilitiesCoreVersion>16.8.0</MicrosoftBuildUtilitiesCoreVersion>
+    <MicrosoftCodeAnalysisFxCopAnalyzersVersion>3.3.1</MicrosoftCodeAnalysisFxCopAnalyzersVersion>
+    <MicrosoftNETTestSdkVersion>16.8.0</MicrosoftNETTestSdkVersion>
+    <NewtonsoftJsonVersion>12.0.3</NewtonsoftJsonVersion>
+    <NUnitVersion>3.12.0</NUnitVersion>
+    <NUnitConsoleRunnerVersion>3.11.1</NUnitConsoleRunnerVersion>
+    <NUnit3TestAdapterVersion>3.17.0</NUnit3TestAdapterVersion>
+    <StyleCopAnalyzersVersion>1.1.118</StyleCopAnalyzersVersion>
+    <SystemCodeDomVersion>5.0.0</SystemCodeDomVersion>
+    <SystemReflectionVersion>4.3.0</SystemReflectionVersion>
+    <SystemReflectionEmitILGenerationVersion>4.7.0</SystemReflectionEmitILGenerationVersion>
+    <SystemReflectionEmitLightweightVersion>4.7.0</SystemReflectionEmitLightweightVersion>
+  </PropertyGroup>
+</Project>
```

Comment:
Makes sense. Should I downgrade some of the libs? e.g. NewtonsoftJson?

Reply:
Yea, let's revert the non-private dependencies back to their original versions.

It's perfectly safe to update the private dependencies (e.g. StyleCop and FxCop) and any unit-testing-only dependencies (e.g. NUnit*).

I created [AVRO-2981](https://issues.apache.org/jira/browse/AVRO-2981) to update the dependencies in Avrogen.

Reply:
1. Do you prefer to keep the version.props way of defining package versions? If yes how does that would work with dependabot? I have never used it, so I have no expereince with it at all.

2. Avro.codegen has no explicit package dependency besides Avro.main, so it will inherit the dependencies of Avro.main. If we keep certain packages at a lower version in Avro.main, but still want to use the latest version in Avro.codegen, we must force to use the latest version by explicitly referencing those packages in Avro.codegen.csproj with the latest version.
If we use versions.props we would need 2 versions for such packages, 1 for the minimum version (e.g. Newtonsoft 10.0.3), and 1 for the latest (e.g. Newtonsoft 12.0.3)
E.g.
Avro.main.csproj: ```<PackageReference Include="Newtonsoft.Json" Version="10.0.3" />```
Avro.codegen.csproj: ```<PackageReference Include="Newtonsoft.Json" Version="12.0.3" />```

Reply:
Avro.codegen.csproj: ```<PackageReference Include="Newtonsoft.Json" Version="*" />``` would work too, however, I am not sure how I feel about usinn an unbound latest version.

Reply:
@blachniet Modded versions.props and the csproj files to support the dependency update strategy. It makes perfect sense what you described with it. Added comments to README and the csproj files as well.